dashboard/lib/services/lti.rb

```diff
@@ -166,10 +166,13 @@ def self.sync_section_roster(lti_integration, lti_section, nrps_members)
   end
 
   # Syncs a course and all its sections from an NRPS response.
-  def self.sync_course_roster(lti_integration:, lti_course:, nrps_sections:, section_owner_id:)
+  def self.sync_course_roster(lti_integration:, lti_course:, nrps_sections:, current_user_id:)
     had_changes = false
     lti_sections = LtiSection.where(lti_course_id: lti_course.id)
 
+    # Only sync the section if there are no sections or the section's owner is equal to the current user
+    return false unless lti_sections.empty? || lti_sections.try(:first).section.user_id == current_user_id
```

Comment:
Is the `.try(:first)` necessary here? Since `lti_sections` is the result of a `.where` it should always have a `first` method.